### ISSUE TITLE
Alternative method for storing mentions

### ISSUE THREAD

**Issue by JordanMussi** [WildcardSearch/MentionMe#62]
At the moment this plugin stores mentions as `@JordanMussi` (or `@"JordanMussi"` or `@'JordanMussi'` etc.). However this leads to querying the DB for a user matching this username. In the first instance this is more costly as usernames with multiple words cause multiple queries, for example `@Jordan Mussi` (discounting the mention cache).
Also `@Jordan Mussi` would match a user called Jordan before myself (annoying :wink:).

My proposal is to store mentions with the user id but only show the username to users.
For example on the MyBB Community Forums my user id is 51459 so typing `@JordanMussi` in a message would be replaced with `<mentionme:51459>` in the DB. 
Furthermore, if I was to change my name previous mentions would not be lost as it is my user id that they are referring to (although the user posting has no idea).

However past mentions would still have to be parsed as before and merging users could create a problem (but the latter is a problem with the current method).

To summarise:
`@JordanMussi`, `@"JordanMussi"`, `@'JordanMussi'` etc. are converted to `<mentionme:51459>` when inserted into the database (in newreply, editpost etc.).

I hope I have made this easy to understand. :wink:
